File: 2000.java
class Solution {
    public String reversePrefix(String word, char ch) {
        int i = word.indexOf(ch);
        if (i != -1) {
            char[] s = word.toCharArray();
            char t;
            int j = 0;
            while (j < i) {
                t = s[i];
                s[i] = s[j];
                s[j] = t;
                j++;
                i--;
            }
            return new String(s);
        }
        return word;
    }
}

File: 2000.py
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        i = 0
        while i < len(word) and word[i] != ch[0]:
            i += 1
        if i == len(word):
            return word
        return word[:i + 1][::-1] + word[i + 1:]
